### PR TITLE
Add startup binary management

### PR DESCRIPTION
**Pull request by torkelrogstad**
On startup, we load the `drivechaind` binary from assets, and launch it. The loading spinner is shown until we get a successful connection. In the case of an error, we display a big and ugly error message that it's impossible to get through. 

TODO: 
- [x] Dynamically load RPC config files based on the CHAIN param
- [x] Add the same thing for sidechains
- [x] ~Load dynamic assets based on platform + architecture~
- [x] Prettify error messages
- [ ] Windows compat
- [x] What to do if there's no config file present. The Drivechain launcher creates it's own config file. I tend to think this is not something we should do without asking the user first, as we should be respectful of the users environment.

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -79,17 +219,38 @@ class SailAppState extends State<SailApp> with WidgetsBindingObserver {
...
           child: SailTheme(
             data: SailThemeData.lightTheme(_sidechain.chain.color),
-            child: const SailScaffold(
+            child: SailScaffold(
               body: Center(
-                child: SailCircularProgressIndicator(),
+                // TODO: better error message with troubleshooting tips here
+                child: _initBinariesError != null
+                    ? SailText.primary24('Ran into unrecoverable error on startup: $_initBinariesError')
+                    : loadingIndicator(),
```
Comment (torkelrogstad) [inline lib/app.dart:253]: @octobocto input appreciated. When this branch is triggered we're at a unrecoverable state, and most likely it's the programmer's fault. Shouldn't happen with the end user, but if it happens they should get a good error message

File: lib/config/sidechains.dart [hunk 2]
```diff
@@ -44,6 +52,72 @@ class EthereumSidechain extends SideWithTicker {
 
   @override
   SidechainType get type => SidechainType.ethereum;
+
+  @override
+  int get rpcPort => 8545;
+
+  @override
+  // TODO: implement binary
+  String get binary => 'sidegeth';
```
Comment (torkelrogstad) [inline lib/config/sidechains.dart:61]: ping

File: lib/pages/tabs/sidechain_explorer_tab_page.dart [hunk 3]
```diff
@@ -82,8 +83,10 @@ class SidechainExplorerTabViewModel extends BaseViewModel {
...
   }
 
-  void setSidechainRPC(SidechainSubRPC sideSubRPC) {
-    _sideRPC.setSubRPC(sideSubRPC);
+  // TODO: this doesn't refresh the UI...
+  void setSidechainRPC(SidechainRPC rpc) {
+    log.d('setting sidechain RPC to "${rpc.chain.name}"');
+    GetIt.I.registerSingleton<SidechainRPC>(rpc);
     notifyListeners();
   }
```
[the remark this answers is not in the document]
Comment (octobocto) [reply lib/pages/tabs/sidechain_explorer_tab_page.dart:91]: You should not re-register it. That does not work.

I ran into the same problem! Check out What I did on the eth-PR. Not the cleanest approach, but it works..

[the remark this answers is not in the document]
Comment (octobocto) [reply lib/pages/tabs/sidechain_explorer_tab_page.dart:91]: By calling the swapChain (or whatever it’s called) the whole view will repaint.

**Comment by octobocto** [comment]
You should really rebase on master before continuing work here…

File: lib/pages/tabs/sidechain_explorer_tab_page.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (torkelrogstad) [reply lib/pages/tabs/sidechain_explorer_tab_page.dart:91]: Thanks!

**Pull request merged**